apps/patient/src/components/statusV2/Header.tsx

```diff
@@ -0,0 +1,240 @@
+import { Box, Heading, HStack, Text, VStack } from '@chakra-ui/react';
+import dayjs from 'dayjs';
+import React from 'react';
+import { CiShop } from 'react-icons/ci';
+import { FiPackage } from 'react-icons/fi';
+import { IoDocumentTextOutline } from 'react-icons/io5';
+import { roundUpTo15MinInterval } from '../../utils/dates';
+import { Step } from './Step';
+import { PrescriptionFulfillment } from '../../__generated__/graphql';
+
+export interface OrderStatusHeaderProps {
+  status: PrescriptionFulfillment['state'] | 'DELAYED' | 'FILLING' | 'SHIPPED';
+  exception?:
+    | 'BACKORDERED'
+    | 'OOS'
+    | 'PA_REQUIRED'
+    | 'PHARMACY_CLOSED'
+    | 'PHARMACY_UNREACHABLE'
+    | 'ORDER_ERROR';
+  pharmacyEstimatedReadyAt?: Date;
+  patientDesiredReadyAt?: Date | 'URGENT';
+}
+
+function headerText(status: OrderStatusHeaderProps['status']) {
+  switch (status) {
+    case 'DELAYED':
+      return 'Order delayed';
+    case 'PROCESSING':
+    case 'RECEIVED':
+    case 'FILLING':
+      return 'Preparing order...';
+    case 'PICKED_UP':
+      return 'Order complete';
+    case 'CREATED':
+    case 'SENT':
+      return 'Order placed';
+    case 'READY':
+      return 'Your order is likely ready';
+    case 'DELIVERED':
+      return 'Order delivered';
+    case 'SHIPPED':
+      return 'Order in transit';
+    default: {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const _: never = status;
```

Comment:
that's cool, so if a new status it should throw an error that it's not handled here?

Reply:
common pattern. At compile time it'll throw a compile error. But at runtime its just code, so you can do whatever you want (usually throw or return an error)